### PR TITLE
Fix map initialization order

### PR DESCRIPTION
**Pull request by atotic**
Fix for #147. Initialize after attached() && apiLoaded.

### CODE REVIEW

File: google-map.html [hunk 1]
```diff
@@ -361,9 +361,18 @@
       }
     },
 
-    _mapApiLoaded: function() {
-      this.map = new google.maps.Map(this.$.map, this._getMapOptions());
+    _initGMap: function() {
+      if (this.map) {
+        return; // already initialized
+      };
+      if (!('google' in window) || !('maps' in window.google)) {
```
Comment (ebidel) [inline google-map.html]: !(window.google && window.google.maps)

File: google-map.html [hunk 2]
```diff
@@ -361,9 +361,18 @@
       }
     },
 
-    _mapApiLoaded: function() {
-      this.map = new google.maps.Map(this.$.map, this._getMapOptions());
+    _initGMap: function() {
+      if (this.map) {
+        return; // already initialized
+      };
```
Comment (ebidel) [inline google-map.html]: remove the `;` on these `if` blocks

File: google-map.html [hunk 3]
```diff
@@ -409,6 +422,7 @@
     },
 
     _updateMarkers: function() {
+      console.log('updateMarkers');
```
Comment (ebidel) [inline google-map.html]: remove

**Comment by ebidel** [comment]
Reviewed

**Comment by atotic** [comment]
fixed. Also added MutationObserver optimizations.

**Comment by atotic** [comment]
PTAL

**Comment by ebidel** [comment]
LGTM. Just the space

**Pull request merged**